From 98ac23cce34e01b3ed6a2ce0ce307ec1fc5dbf3c Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Thu, 3 Mar 2016 17:44:33 -0500
Subject: [PATCH 1196/1565] drm/amd/dal: fix reset mode warning msg

We should call surface programming in case of mode
reset (actual reset or DPMS). Warning was printed
that target is NULL.

Also removed debug prints for buffers prepare/remove,
that occurs on each pageflip and pollutes log

Change-Id: I96d6bdf63be5c08bad3057c9cd825afbcc9cb245
Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index b91eced..5f35ab4 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -1430,7 +1430,6 @@ static int dm_plane_helper_prepare_fb(
 
 	afb = to_amdgpu_framebuffer(new_state->fb);
 
-	DRM_DEBUG_KMS("Pin new framebuffer: %p\n", afb);
 	obj = afb->obj;
 	rbo = gem_to_amdgpu_bo(obj);
 	r = amdgpu_bo_reserve(rbo, false);
@@ -1461,7 +1460,6 @@ static void dm_plane_helper_cleanup_fb(
 		return;
 
 	afb = to_amdgpu_framebuffer(old_state->fb);
-	DRM_DEBUG_KMS("Unpin old framebuffer: %p\n", afb);
 	rbo = gem_to_amdgpu_bo(afb->obj);
 	r = amdgpu_bo_reserve(rbo, false);
 	if (unlikely(r)) {
@@ -2272,7 +2270,8 @@ int amdgpu_dm_atomic_commit(
 		struct drm_connector *connector;
 		struct dm_connector_state *dm_state = NULL;
 
-		if (!fb || !crtc || !crtc->state->planes_changed)
+		if (!fb || !crtc || !crtc->state->planes_changed ||
+			!crtc->state->active)
 			continue;
 
 		if (page_flip_needed(
-- 
1.9.1

